### PR TITLE
Add SNI support in doc

### LINKED ISSUE #2450: Change required: connection errors
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

Clarify SNI support 
https://docs.risingwave.com/docs/current/python-client-libraries/#use-sqlalchemy-to-connect-to-risingwave  https://neon.tech/docs/connect/connection-errors

### Reference

_No response_

### PR DESCRIPTION
**Pull request by WanYixian**
## Info

- **Description**

  - Add SNI support in doc, see preview below, let me know if any comments! 
image

- **Related doc issue**
  
  Resolves #2450 

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

**Review by Ad-Bean — APPROVED** [review]
LGTM! cc @mikechesterwang

File: cloud/project-connection-errors.md [hunk 1]
```diff
@@ -65,6 +65,18 @@ curl -L --create-dirs -o $HOME/.postgresql/root.crt 'https://risingwave.cloud/ro
...
 
+## The endpoint ID is not specified
+
+This error occurs when your Postgres client doesn't support Server Name Indication (SNI), so it can't automatically send the endpoint ID needed to route your request correctly.
+
+RisingWave Cloud has tested the following drivers for SNI support:
+
+| Driver               | Language | SNI Support | Notes                                      |
+|----------------------|----------|-------------|--------------------------------------------|
+| psycopg2                |   Python     | ✓            |    Requires the `'sslmode': 'require'` option                                        |
```
Comment (ShanlanLi) [inline cloud/project-connection-errors.md]:
```suggestion
| psycopg2                |   Python     | ✓            |    Requires the `'sslmode': 'require'` option.                                        |
```

Comment (ShanlanLi) [reply cloud/project-connection-errors.md]: Does it need punctuation?

**Review by ShanlanLi — APPROVED** [review]
LGTM

File: cloud/project-connection-errors.md (hunk 1 above)
Comment (WanYixian) [reply cloud/project-connection-errors.md]: Can omit punctuation for short phrase like this in table

**Comment by cloudcarver** [comment]
what's the `endpoint ID`? Do we have this concept now? Is it `tenant identifier`?

**Comment by Ad-Bean** [comment]
> what's the `endpoint ID`? Do we have this concept now? Is it `tenant identifier`?

will change to `tenant identifier`

**Comment by WanYixian** [comment]
> what's the `endpoint ID`? Do we have this concept now? Is it `tenant identifier`?

Updated, PTAL, thanks!

**Review by cloudcarver — APPROVED** [review]
LGTM, please add more libraries. At least cover those listed in the portal, like `node-postgres` , `JDBC`, `pgx/v4`, `libpq` (for `libpq`, we need to specify the version of `psql`, only the newer versions support SNI).

**Pull request merged**